applications/Chat/evaluate/gpt_evaluate.py

```diff
@@ -0,0 +1,525 @@
+    metrics = prompt["metrics"]
+    inst["evaluation"] = {}
+
+    for metric in metrics.keys():
+        for i in range(MAX_API_RETRY):
+            try:
+                response = openai.Completion.create(
+                    model="text-davinci-003",
+                    prompt=prompt["prompt"].format(
+                        question=question,
+                        answer=answer,
+                        metric=prompt["metrics"][metric],
+                        steps=prompt["CoT"][metric],
+                    ),
+                    logprobs=5,
+                    temperature=0,
+                    max_tokens=max_tokens,
+                )
+                inst["evaluation"][metric] = {
+                    "response": response["choices"][0]["text"],
+                    "logprobs": response["choices"][0]["logprobs"]["top_logprobs"],
+                }
+                break
+            except Exception as e:
+                print(e)
+                time.sleep(1)
+    return inst
+
+
+def gpt35_evaluate(
+    answers: List[Dict],
+    prompts: List[Dict],
+    model_name: str,
+    ignore_categories: List[str],
+    save_path: str,
+) -> Dict[str, Any]:
+    """
+    Use GPT-3.5 to evaluate model answers and save evaluation results.
+
+    Args:
+        answers: model answers.
+        prompts: prompts for all categories.
+        model_name: name of the model.
+        ignore_categories: the categories that should be ignored. Some categories can be evaluated by automatic metrics.
+        save_path: path to save GPT-3.5 evaluations.
+
+    Returns:
+        All the evaluations of the given answers.
+    """
+
+    prompt_per_category = {prompt["category"]: prompt for prompt in prompts}
+
+    data_per_category = {}
+    for answer in answers:
+        category = answer["category"]
+
+        if category in ignore_categories:
+            continue
+
+        if answer["category"] in data_per_category.keys():
+            data_per_category[category].append(answer)
+        else:
+            data_per_category[category] = [answer]
+
+    categories_str = ", ".join(x for x in list(data_per_category.keys()))
+    print(f"The evaluated categories are {categories_str}.")
+
+    if not os.path.exists(save_path):
+        os.makedirs(save_path)
+
+    all_evaluations = []
+
+    for category, data in data_per_category.items():
+        if prompt_per_category.get(category) == None:
+            print(f"No metrics for category {category}! Use category general now.")
+            prompt_per_category[category] = prompt_per_category["general"]
+
+        if os.path.exists(os.path.join(save_path, model_name, f"{category}_evaluation_results.json")):
+            evaluations = jload(os.path.join(save_path, model_name, f"{category}_evaluation_results.json"))
+            all_evaluations.extend(evaluations)
+            print(f"Already finished evaluating category {category}! Skip evaluation for this category.")
+            continue
```

Comment:
What if we change some answers and want to get the new evaluation results? Do we need to remove all these files?